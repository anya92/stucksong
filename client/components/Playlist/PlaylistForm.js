import React from 'react';
import { Link } from 'react-router-dom';

import {
  PlaylistContainer,
  FormGroup,
  InputGroup,
  InputGroupAddon,
  DangerInfo,
  FormButton,
} from '../../styles/playlist';

const PlaylistForm = ({
  name,
  description,
  numberOfTracks,
  handleChange,
  createPlaylist,
}) => (
  <PlaylistContainer>
    <h1>Create a playlist of your <Link to="/top-tracks">Top&nbsp;Tracks</Link></h1>
    <form onSubmit={e => createPlaylist(e)}>
      <FormGroup>
        <label htmlFor="playlist-name">Name
          <InputGroup>
            <input
              id="playlist-name"
              type="text"
              maxLength="100"
              value={name}
              onChange={e => handleChange('name', e.target.value)}
            />
            <InputGroupAddon>{name.length}/100</InputGroupAddon>
          </InputGroup>
        </label>
      </FormGroup>
      <FormGroup>
        <label htmlFor="playlist-description">Description
          <InputGroup>
            <textarea
              id="playlist-description"
              rows={3}
              maxLength="300"
              placeholder="Give your playlist a description."
              value={description}
              onChange={e => handleChange('description', e.target.value)}
            />
            <InputGroupAddon>
              {description.length}/300
            </InputGroupAddon>
          </InputGroup>
        </label>
      </FormGroup>
      <FormGroup number>
        <label htmlFor="playlist-number-of-tracks">Number of Tracks
          <input
            id="playlist-number-of-tracks"
            type="number"
            className={`${numberOfTracks < 1 || numberOfTracks > 50 ? 'danger' : ''}`}
            min={1}
            max={50}
            value={numberOfTracks}
            onChange={e => handleChange('numberOfTracks', e.target.value)}
          />
          {
            (numberOfTracks < 1 || numberOfTracks > 50)
            && <DangerInfo>Number of tracks must be between 1  and 50.</DangerInfo>
          }
        </label>
      </FormGroup>
      <FormButton>
        <button type="submit">Create</button>
      </FormButton>
    </form>
  </PlaylistContainer>
);

export default PlaylistForm;
