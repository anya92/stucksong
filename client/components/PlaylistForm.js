import React from 'react';
import { Link } from 'react-router-dom';

import {
  PlaylistContainer,
  FormGroup,
  InputGroup,
  InputGroupAddon,
  DangerInfo,
  FormButton,
} from '../styles/playlist';

const PlaylistForm = ({
  name,
  description,
  numberOfTracks,
  handleChange,
  createPlaylist,
}) => (
  <PlaylistContainer>
    <h1>Create a playlist of your <Link to="/top-tracks">Top&nbsp;Tracks</Link></h1>
    <form onSubmit={e => createPlaylist(e)}>
      <FormGroup>
        <label>Name</label>
        <InputGroup>
          <input 
            type="text"
            maxLength="100"
            value={name}
            onChange={e => handleChange('name', e.target.value)}
          />
          <InputGroupAddon>{name.length}/100</InputGroupAddon>
        </InputGroup>
      </FormGroup>
      <FormGroup>
        <label>Description</label>
        <InputGroup>
          <textarea
            rows={3}
            maxLength="300"
            placeholder="Give your playlist a description."
            value={description}
            onChange={e => handleChange('description', e.target.value)}
          />
          <InputGroupAddon>{description.length}/300</InputGroupAddon>
        </InputGroup>
      </FormGroup>
      <FormGroup number>
        <label>Number of Tracks</label>
        <input
          type="number"
          className={`${numberOfTracks < 1 || numberOfTracks > 50 ? 'danger' : ''}`}
          min={1}
          max={50}
          value={numberOfTracks}
          onChange={e => handleChange('numberOfTracks', e.target.value)}
        />
        {
          (numberOfTracks < 1 || numberOfTracks > 50)
          && <DangerInfo>Number of tracks must be between 1  and 50.</DangerInfo>
        }
      </FormGroup>
      <FormButton>
        <button type="submit">Create</button>
      </FormButton>
    </form>
  </PlaylistContainer>
);

export default PlaylistForm;
