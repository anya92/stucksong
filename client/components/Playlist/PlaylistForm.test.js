import React from 'react';
import { shallow } from 'enzyme';

import PlaylistForm from './PlaylistForm';

describe('PlaylistForm component', () => {
  let wrapper;
  let handleChange;
  let createPlaylist;

  beforeEach(() => {
    handleChange = jest.fn();
    createPlaylist = jest.fn();
    const mockProps = {
      name: '',
      description: '',
      numberOfTracks: 25,
      handleChange,
      createPlaylist,
    };
    wrapper = shallow(<PlaylistForm {...mockProps} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls handleChange on name change', () => {
    const e = {
      target: {
        value: 'Top 10',
      },
    };
    wrapper.find('#playlist-name').simulate('change', e);
    expect(handleChange).toHaveBeenLastCalledWith('name', 'Top 10');
  });

  it('calls handleChange on description change', () => {
    const e = {
      target: {
        value: 'My top 10 songs',
      },
    };
    wrapper.find('#playlist-description').simulate('change', e);
    expect(handleChange).toHaveBeenLastCalledWith('description', 'My top 10 songs');
  });

  it('calls handleChange on number of tracks change', () => {
    const e = {
      target: {
        value: 10,
      },
    };
    wrapper.find('#playlist-number-of-tracks').simulate('change', e);
    expect(handleChange).toHaveBeenLastCalledWith('numberOfTracks', 10);
  });

  it('calls createPlaylist on form submit', () => {
    wrapper.find('form').simulate('submit');
    expect(createPlaylist).toHaveBeenCalled();
  });
});
