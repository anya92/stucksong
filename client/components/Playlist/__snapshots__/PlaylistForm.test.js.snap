// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistForm component renders correctly 1`] = `
<glamorous(div)>
  <h1>
    Create a playlist of your 
    <Link
      replace={false}
      to="/top-tracks"
    >
      TopÂ Tracks
    </Link>
  </h1>
  <form
    onSubmit={[Function]}
  >
    <glamorous(div)>
      <label
        htmlFor="playlist-name"
      >
        Name
        <glamorous(div)>
          <input
            id="playlist-name"
            maxLength="100"
            onChange={[Function]}
            type="text"
            value=""
          />
          <glamorous(div)>
            0
            /100
          </glamorous(div)>
        </glamorous(div)>
      </label>
    </glamorous(div)>
    <glamorous(div)>
      <label
        htmlFor="playlist-description"
      >
        Description
        <glamorous(div)>
          <textarea
            id="playlist-description"
            maxLength="300"
            onChange={[Function]}
            placeholder="Give your playlist a description."
            rows={3}
            value=""
          />
          <glamorous(div)>
            0
            /300
          </glamorous(div)>
        </glamorous(div)>
      </label>
    </glamorous(div)>
    <glamorous(div)
      number={true}
    >
      <label
        htmlFor="playlist-number-of-tracks"
      >
        Number of Tracks
        <input
          className=""
          id="playlist-number-of-tracks"
          max={50}
          min={1}
          onChange={[Function]}
          type="number"
          value={25}
        />
      </label>
    </glamorous(div)>
    <glamorous(div)>
      <button
        type="submit"
      >
        Create
      </button>
    </glamorous(div)>
  </form>
</glamorous(div)>
`;
